### ISSUE TITLE
Unable to run queries of some of the hudi tables using Hudi connector after trino 418

### ISSUE THREAD

**Issue by ryadav-uptycs** [trinodb/trino#19975]
Hi All,
I have recently upgraded from **trino 418 to trino 430** and found that hudi-connector is not working as expected for some of our hudi tables
we found 2 major issues:
1. **it is show incorrect table count for some of the tables where we run clustering**
    - After digging deeper into the code we found that in newer version of trino .hudi metapath listing is done recursively (https://github.com/trinodb/trino/blob/430/plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/table/HudiTableMetaClient.java#L166) which is causing to incorrect oldest timestamp  file calcaution
      .Since it is doing recurvely it is  also reading commit files  from .hoodie/.bucket_index/consistent_hashing_metadata/<part1>/<part2>20231101000038468.commit  which was placed to update the state of clustering . it is causing to skip some of the parquet files while listing the directory . hence incorrectly data count. This clustering commit files in hudi was done as part of this PR (https://github.com/apache/hudi/pull/8503)

2. **For some of the hudi tables even simple select * from table limit 1 is hanging and not giving any error or output.**
   - while debugging the issue , while listing the parition it launching multiple thread and if any thread has any error then the exception is  not captured properly in the caller thread which is causing it to going into hanging state. (https://github.com/trinodb/trino/blob/430/plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/HudiSplitSource.java#L102-L103) in this trinoException  value is checked to handle exception but it is never set , which was not the case with trino-418

   - Also , after debugging what has caused the issue , found out that if the partition directory is empty then it tries to create the dir but since dir already exists it is throwing dir already exists 
     error. (https://github.com/trinodb/trino/blob/430/plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/table/HudiTableFileSystemView.java#L300)

After fixing all the above mentioned issue in my local git it is working as expected .

**Comment by ebyhr** [comment]
> After fixing all the above mentioned issue in my local git it is working as expected .

Thanks for reporting this issue. Could you submit the PR?

**Comment by rajneshbhatia-dglobal** [comment]
is there any solution to point 2 ?

**Comment by ryadav-uptycs** [comment]
> is there any solution to point 2 ?

@rajneshbhatia-dglobal  yes , it occurred because there was an empty partition in my case and existing logic try to create the parition path if it is empty and it is failing because path was already there but it is empty. This failure was not handled properly which is causing in to fail.                                                                                                                                                                                                                     
For this , I have raised a PR -> https://github.com/trinodb/trino/pull/20147  
and modified the plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/HudiSplitSource.java class to handle exception
and plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/table/HudiTableFileSystemView.java  -> to avoid creating dir

**Comment by lordicecream** [comment]
Has there been any fix for this? We are still facing this issue with version 433

**Comment by ryadav-uptycs** [comment]
@lordicecream  I have raised the PR for this
